pkg/scheduler/core/generic_scheduler.go

```diff
@@ -211,12 +211,15 @@ func (g *genericScheduler) selectHost(nodeScoreList framework.NodeScoreList) (st
 
 // numFeasibleNodesToFind returns the number of feasible nodes that once found, the scheduler stops
 // its search for more feasible nodes.
-func (g *genericScheduler) numFeasibleNodesToFind(numAllNodes int32) (numNodes int32) {
-	if numAllNodes < minFeasibleNodesToFind || g.percentageOfNodesToScore >= 100 {
+func (g *genericScheduler) numFeasibleNodesToFind(numAllNodes int32, prof *profile.Profile) (numNodes int32) {
+	adaptivePercentage := g.percentageOfNodesToScore
+	if prof.PercentageOfNodesToScore != 0 {
+		adaptivePercentage = prof.PercentageOfNodesToScore
+	}
```

Comment:
should probably make a note in the godoc for the API types that profile score percentages will override the scheduler-level setting. I think it's a good idea to do that, worth noting